controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -944,118 +944,158 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
 
     private CompletableFuture<Map<Long, Long>> getTruncationStreamCutBySizeLimit(String scope, String stream, OperationContext context, RetentionPolicy policy,
                                                                                  RetentionSet retentionSet, Map<Long, Long> lowerBound) {
-        // if the lowerbound on subscribers streamcuts satisfies the policy size bound, then return it. 
-        // else return the stream cut that satisfies maximum bound on size. 
         // 1. if lowerbound.size < max and lowerbound.size > min truncate at lowerbound
-        // 2. if lowerbound.size < min, truncate at max irrespective of if lowerbound overlaps with max or not. 
+        // 2. if lowerbound.size < min, truncate at streamcut behind lowerbound that satisfies the policy. (we look at max and below).
```

Comment:
Can't seem to wrap my head easily around terms "behind" and "forward" here... 
Could we instead reference everything w.r.t the tail of the Stream?
If `lowerbound.size < min` I thought we should be truncating at a stream-cut closest-to the `min` but with size > min. (further from the tail than min)

Reply:
replaced behind with less than and forward with greater than.. the terminology we use elsewhere

Reply:
>  Could we instead reference everything w.r.t the tail of the Stream?
how do you mean? 

basically we compare two streamcuts to determine if one streamcut is greater than or less than the other.. 
greater than means it points to more recent data. less than means older data.. overlapping means its pointing to some older data and some newer data.